### DISCUSSION TITLE
Are the generated Rust files platform-independent?

### DISCUSSION THREAD

**Discussion by hug-dev** [tokio-rs/prost#487] [Q&A]
Hello 👋!

Thanks for the crate, we are using it our Parsec project and really liking it.
This is more a question than a problem. To remove the dependency on `protoc` at build time, we were thinking of pre-generating all Rust files from our Protobuf files (using `prost_build`) and storing them in tree ([issue](https://github.com/parallaxsecond/parsec-interface-rs/issues/98) for reference).

The main question if we do that is if we need to generate one set of files per Rust target or if that's fine just doing it once for all targets.

To me it seems that the generated file would not contain any platform specific code so that would be ok. I was not totally sure so wondering if anyone would know more about that!

**Comment by carllerche** [comment]
The generated files should be platform-independent.
